presto-native-execution/presto_cpp/main/operators/BroadcastFactory.cpp

```diff
@@ -106,8 +111,21 @@ RowVectorPtr BroadcastFileWriter::fileStats() {
       std::vector<VectorPtr>({std::move(data)}));
 }
 
+void BroadcastFileWriter::initializeWriteFile() {
+  if (!writeFile_) {
+    LOG(INFO) << "Opening broadcast file for write: " << filename_;
+    writeFile_ = fileSystem_->openFileForWrite(filename_);
+  }
+}
+
 void BroadcastFileWriter::serialize(const RowVectorPtr& rowVector) {
   auto numRows = rowVector->size();
+  if (numRows == 0) {
+    return;
+  }
+
+  initializeWriteFile();
```

Comment:
Initiating write file in a serialize method seems odd.
Better to call this method `write`.

Reply:
good point! updated